.github/workflows/ci.yaml

```diff
@@ -1,73 +1,58 @@
 name: ci
 on:
   push:
```

Comment:
We really want to run test and lint on every push to every branch, even when there's no PR open? Seems like a waste to me. At the least, we should keep the scheduled runs since we're not fulling pinning the build matrix.

Reply:
This is our standard for other repos. Unfortunately there's no [pull_request type](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) that includes push on the PR branch - so users would have to click some random buttons (eg add a label) to actually get the tests to rerun, unfortunately. This matches our internal configs.

Reply:
More of a drive by, but doesn't this do what you want?

```
on:
  push:
    branches:
      - main
  pull_request:
    types: [opened, synchronize, reopened]
```

Always run on push to main, and run whenever a PR is opened, reopened or synchronized.

Synchronized is defined as:

>  ... push a new commit to the HEAD ref of a pull request then this “synchronize” event will be triggered. This is because the system is syncing the pull requests with the latest changes. This will NOT trigger if the base ref is updated.

Reply:
🤦 I guess I don't know the meaning of `synchronize`. Thanks for the drive-by comment!